### PR TITLE
hack: remove -installsuffix build flag

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/moby/moby/pull/44546#discussion_r1054961381

**- What I did**

`-installsuffix netgo` has been added in https://github.com/moby/moby/pull/10155 to work around a bug with "go build" but not required anymore since go 1.5: https://github.com/golang/go/commit/4dab6d01f12591f256d36b32cd6480ef679458f1

Also took the opportunity to:
* remove `ORIG_BUILDFLAGS` var that was used for the cross target but has been removed
in https://github.com/moby/moby/pull/44546 so not necessary anymore
* cleanup unnecessary vars in `make.sh` script
* display build cmd when `DOCKER_DEBUG` is set (had to remove extra tabs and new lines to avoid a clunky output)

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: hack/make/.go-autogen [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-	-X \"github.com/docker/docker/dockerversion.PlatformName=${PLATFORM}\" \
-	-X \"github.com/docker/docker/dockerversion.ProductName=${PRODUCT}\" \
-	-X \"github.com/docker/docker/dockerversion.DefaultProductLicense=${DEFAULT_PRODUCT_LICENSE}\" \
-"
+-X \"github.com/docker/docker/dockerversion.Version=${VERSION}\" \
+-X \"github.com/docker/docker/dockerversion.GitCommit=${GITCOMMIT}\" \
+-X \"github.com/docker/docker/dockerversion.BuildTime=${BUILDTIME}\" \
+-X \"github.com/docker/docker/dockerversion.PlatformName=${PLATFORM}\" \
+-X \"github.com/docker/docker/dockerversion.ProductName=${PRODUCT}\" \
+-X \"github.com/docker/docker/dockerversion.DefaultProductLicense=${DEFAULT_PRODUCT_LICENSE}\" "
```
Comment (thaJeztah) [inline hack/make/.go-autogen:9]: Could be worth keeping the last quote on a separate line (it's easy to miss that it's there); probably that was one of the reasons to use the indentation here.

**Review by thaJeztah — COMMENTED** [review]
Code changes look good to me (left one minor comment).


Also looked at the commit messages, and I know referring to PR numbers can be "noisy"; I generally use commits instead (which also allows finding back the related change without GitHub). Perhaps you could change those?

```
hack: remove -installsuffix build flag

Has been introduced in #10155
to work around a bug with "go build" but not required anymore
```

Can you change the PR number to `232d59baeb13778abc242a602ca434d83e1eb6e8` (commit?)

```
hack: remove ORIG_BUILDFLAGS var
This var was used for the cross target but it has been removed
in #44546 so not necessary anymore
```

Can you change the PR number to a commit? (Is there a specific commit we can point to? Otherwise, we could use the merge commit; `b9fe30dad49db41fc8fb41da8fef46f0fb0d42b9`

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by corhere — APPROVED** [review]

**Comment by thaJeztah** [comment]
Let me bring this one in. I added a cherry-pick label to keep the 23 branch a bit cleaned up as well, but it's definitely not an urgent one.

**Pull request merged**